js/PulsarFormComponent.js

```diff
@@ -84,33 +87,56 @@ PulsarFormComponent.prototype.initTimePickers = function () {
  */
 PulsarFormComponent.prototype.initDatePickers = function () {
     const datepickers = this.$html.find('[data-datepicker="true"]');
-    let defaultDateFormat = 'DD/MM/YYYY';
 
     datepickers.each((index, element) => {
-        let dateFormat = element.getAttribute('data-format');
+        let $element = $(element);
+        let dateFormat;
+        let formatKey = element.getAttribute('data-format');
+        let locale = element.getAttribute('data-locale');
 
-        // Check if data-format attribute exists and lowercase it
-        // to eliminate different styles of writing issues
-        if (dateFormat !== null) {
-            dateFormat = dateFormat.toLowerCase();
+        if (typeof locale === 'string') {
+            dateFormat = moment.localeData(locale).longDateFormat('L');
         }
 
-        switch (dateFormat) {
-            case 'us':
-                defaultDateFormat = 'MM/DD/YYYY';
-                break;
-            case 'reverse':
-                defaultDateFormat = 'YYYY/MM/DD';
-                break;
-            default:
-                defaultDateFormat = 'DD/MM/YYYY';
+        if (typeof formatKey === 'string') {
+            formatKey = formatKey.toLowerCase();
+
+            switch (formatKey) {
+                case 'us':
+                    dateFormat = 'MM/DD/YYYY';
+                    break;
+                case 'reverse':
+                    dateFormat = 'YYYY/MM/DD';
+                    break;
+                default:
+                    dateFormat = 'DD/MM/YYYY';
+            }
         }
 
         // Initialize pikaday with the correct date format
-        $(element).pikaday({ format: defaultDateFormat });
+        $element.pikaday({ format: dateFormat });
```

Comment:
Can you add a check in here, while the helper will now always set a `data-locale` attribute there may be hardcoded HTML date pickers in production which won't have either `data-format` or `data-locale` which may result in `dateFormat` being undefined, if I'm not mistaken?

Reply:
👍 Have made the default the UK format which can be changed via the attributes